packages/lib/src/core/Analytics/AnalyticsEvent.ts

```diff
@@ -0,0 +1,13 @@
+import uuid from '../../utils/uuid';
+import { getUTCTimestamp } from './utils';
+
+export abstract class AnalyticsEvent {
+    public timestamp: string;
+    public id: string;
+    public component: string;
+
```

Comment:
I believe `target` can also be added here in the abstract object. It is available for all event types

Reply:
No, it's not allowed on an error event